decision-process/README.md

```diff
@@ -0,0 +1,77 @@
+---
+type: decision
+stakeholders:
+    - sig-architecture
+    - team-horizon
+    - team-team
+issue:
+---
+
+# RFC Decision Making Process
+
+This RFC follows on from the RFC repository README (see [here](https://github.com/giantswarm/rfc/blob/b3f4d2af6ff8965e542baae7d0e8fadb9163f356/README.md)).
+
+This RFC aims to expand on the current RFC process by adding a lightweight decision making process.
+The aim of this decision making process is to help enable, improve, and streamline technical decision making at Giant Swarm.
+
+## RFC Types
+
+To allow for a decision making process without removing the current free-form nature of RFCs, RFC types are proposed. The following changes will be introduced:
+
+- All RFCs will have a YAML header (front matter)
+- The YAML header will contain the following mandatory fields:
+    - `type`
```

Comment:
Date should be required, and since it should be easy to change/fix the date, it should go into the YAML header and not into the directory name. And there could be multiple dates such as `created: 2023-08-30`, `approved: 2023-09-24`, `deprecated: 2024-05-10`. Some companies including us even tried `must_be_reviewed_until`, but that attempt will fail until we have an actual process in place that reminds the reviewer group automatically, and even then time pressure is very dependent on the case and may still be a failure, so I wouldn't recommend this initially.

Some CI job should ensure correctness, and if a new field is introduced later on, could align the existing files automatically.

There could also be fields that relate RFCs to the others, such as `followup_for: [ ... ]`, `related: [ ... ]` or `replaces: [...]` (newer version of a deprecated decision). This would make it easier to render the information in a structured way, rather than people writing "Further reading" bullet-point lists in free form somewhere in the "article", impossible to parse.